CharlieBackend.Business/Services/Interfaces/IStudentGroupService.cs

```diff
@@ -10,7 +10,7 @@ public interface IStudentGroupService
     {
         Task<Result<StudentGroupDto>> CreateStudentGroupAsync(CreateStudentGroupDto studentGroupModel);
 
-        Task<IList<StudentGroupDto>> GetAllStudentGroupsAsync();
+        Task<Result<IList<StudentGroupDto>>> GetAllStudentGroupsAsync();
```

Comment:
Don't forget to call .ToActionResult in controller action. Otherwise, it will change schema of response body.